[Dion Neo] iP

noobmaster19 opened the pull request:
hi

noobmaster19 committed:
'level1'

noobmaster19 committed:
'level2'

noobmaster19 committed:
'level3'

noobmaster19 committed:
'level4

noobmaster19 committed:
'level5'

noobmaster19 committed:
'level6'

noobmaster19 committed:
'test-ui-test'

noobmaster19 committed:
'init'

noobmaster19 committed:
'test'

noobmaster19 committed:
'init'

noobmaster19 committed:
'merged'

noobmaster19 committed:
'oop'

noobmaster19 committed:
Here i did abit of refactoring as well.

noobmaster19 committed:
'jar-ed'

noobmaster19 committed:
'init'

noobmaster19 committed:
'init'

noobmaster19 committed:
Merge branch 'Level-9'

noobmaster19 committed:
Changed README.md to follow GFMD

noobmaster19 committed:
'gradle-support-added'

tsm1820 commented on lib/src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,130 @@
+package duke.utils;
+
+import java.util.*;
```
Overall everything looks good, but perhaps can use specific import statements instead of using wildcard imports?

tsm1820 commented on lib/src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,130 @@
+
+import duke.tasks.*;
+
+import java.io.*;
```
Same point regarding wildcard imports as above :)

tsm1820 reviewed (commented):
LGTM. Just the bits regarding the import statements.

Neha-5678 commented on lib/src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import java.util.*;
```
Maybe you could consider explicitly stating the imported classes? :)

Neha-5678 commented on lib/src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,130 @@
+            while (s.hasNext()) {
+                String[] parsed = s.nextLine().split(",");
+                String type = parsed[0];
+                boolean done = (parsed[1].equals("1"));
```
Maybe it would be clearer if you changed the name of the boolean variable to isDone? Just a suggestion!

Neha-5678 commented on lib/src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,130 @@
+            Scanner s = new Scanner(data);
+            while (s.hasNext()) {
+                String[] parsed = s.nextLine().split(",");
+                String type = parsed[0];
```
This is a very minute detail but maybe you could consider renaming the variable to make it clearer? for instance, typeOfTask?

Neha-5678 commented on lib/src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,130 @@
+                String type = parsed[0];
+                boolean done = (parsed[1].equals("1"));
+                if (type.equals("todo")) {
+                    savedTasks.add(new Todo(parsed[2], done));
```
I was thinking that maybe it would be better if you had separate variables declared for parsed[2] and parsed[3] so that it is clearer what you are passing into the constructor methods? Again, this is just a suggestion! :)

Neha-5678 commented on lib/src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+            Task task = userInputs.get(taskIndex - 1);
+            userInputs.remove(taskIndex - 1);
+            System.out.println(" " + task);
+            task = null;
```
Very trivial but I was just thinking if there is a need to set the task variable back to null since you won't be using it elsewhere?

Neha-5678 reviewed (commented):
good job! Overall, a well-written code and I could only spot the minute details.

noobmaster19 committed:
gui-implemented

noobmaster19 committed:
gradle-support

noobmaster19 committed:
removed-pictures

noobmaster19 committed:
Subject: Removed lib folder

There was an unneccassry lib folder that was placed there by mistake. It will be removed in this commmit.

noobmaster19 committed:
assertions-added

noobmaster19 committed:
Refactored flow of parsing integers

Previously, integers were parsed at tasklist level. DeleteCommand and DoneCommand stores index as strings. This can be quite
confusing, so i have refactored it to do the parsing and validation on the command level instead.

noobmaster19 committed:
Merge pull request #1 from noobmaster19/A-Assertions

assertions-added

noobmaster19 committed:
merged

noobmaster19 committed:
s

noobmaster19 committed:
Merge commit 'd4229a6'

noobmaster19 committed:
renamed

noobmaster19 committed:
link-fixed

noobmaster19 committed:
s

noobmaster19 committed:
fixed-imports

noobmaster19 committed:
fixed-jar

noobmaster19 committed:
Added more header comments for class and methods

noobmaster19 committed:
removed-ui

noobmaster19 committed:
Changed catch all exception to be more specific
- now all exceptions for accessing mem is caught at command exc level
- removed all catch all exceptions
- More user-friendly messages

noobmaster19 committed:
fixed-exceptions
